bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/connector/EpsonProjectorTcpConnector.java

```diff
@@ -0,0 +1,170 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Connector for TCP communication.
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class EpsonProjectorTcpConnector implements EpsonProjectorConnector {
+
+    private static final Logger logger = LoggerFactory.getLogger(EpsonProjectorTcpConnector.class);
+
+    String ip = null;
+    int port = 10000;
+    Socket socket = null;
+    InputStream in = null;
+    OutputStream out = null;
+
+    public EpsonProjectorTcpConnector(String ip, int port) {
+        this.ip = ip;
+        this.port = port;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void connect() throws EpsonProjectorException {
+        logger.debug("Open connection to address'{}:{}'", ip, port);
+
+        try {
+            socket = new Socket(ip, port);
+            in = socket.getInputStream();
+            out = socket.getOutputStream();
+        } catch (Exception e) {
+            throw new EpsonProjectorException(e);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void disconnect() throws EpsonProjectorException {
+        if (out != null) {
+            logger.debug("Close tcp out stream");
+            IOUtils.closeQuietly(out);
+        }
+        if (in != null) {
+            logger.debug("Close tcp in stream");
+            IOUtils.closeQuietly(in);
+        }
+        if (socket != null) {
+            logger.debug("Closing socket");
+            try {
+                socket.close();
+            } catch (IOException e) {
+                logger.warn("Error occurred when closing tcp socket", e);
+            }
+        }
+
+        socket = null;
+        out = null;
+        in = null;
+
+        logger.debug("Closed");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String sendMessage(String data, int timeout) throws EpsonProjectorException {
+        if (in == null || out == null) {
+            connect();
+        }
+
+        try {
+            // flush input stream
+            if (in.markSupported()) {
+                in.reset();
+            } else {
+
+                while (in.available() > 0) {
+
+                    int availableBytes = in.available();
+
+                    if (availableBytes > 0) {
+
+                        byte[] tmpData = new byte[availableBytes];
+                        in.read(tmpData, 0, availableBytes);
+                    }
+                }
+            }
+
+            return sendMmsg(data, timeout);
+
+        } catch (IOException e) {
+
+            logger.debug("IO error occurred...reconnect and resend ones");
+            disconnect();
+            connect();
+
+            try {
+                return sendMmsg(data, timeout);
+            } catch (IOException e1) {
+                throw new EpsonProjectorException(e);
+            }
+
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception